frontend/src/bookings/calendar.vue

```diff
@@ -0,0 +1,197 @@
+  import 'fullcalendar/dist/fullcalendar.css'
+  import 'fullcalendar-scheduler'
+
+  export default {
+    name: 'Calendar',
+    created() {
+      this.getExams()
+      this.getBookings()
+      this.getRooms()
+    },
+    mounted() {
+      this.$root.$on('next', () => { this.next() })
+      this.$root.$on('prev', () => { this.prev() })
+      this.$root.$on('today', () => { this.today() })
+      this.$root.$on('month', () => { this.month() })
+      this.$root.$on('listWeek', () => { this.listWeek() })
+      this.$root.$on('listDay', () => { this.listDay() })
+    },
+    data() {
+      return {
+        config: {
+          schedulerLicenseKey: 'GPL-My-Project-Is-Open-Source',
+          showNonCurrentDates: false,
+          fixedWeekCount: false,
+          navLinks: true,
+          defaultView: 'listWeek',
+          views: {
+            listDay: {
+              allDaySlot: false,
+            },
+            listWeek: {
+              allDaySlot: false,
+            },
+          },
+          height() {
+            return window.innerHeight - 230
+          },
+          weekends: false,
+          maxTime: '18:00:00',
+          minTime: '8:00:00',
+          header: {
+            left: null,
+            center: null,
+            right: null
+          },
+          groupByDateAndResource: true
+        },
+        title: '',
+      }
+    },
+    computed: {
+      ...mapState([
+        'calendarScheduleMode',
+          'eventsSource',
+          'resources',
+          'bookings',
+          'rooms',
+          'exams'
+      ]),
+      events() {
+        return this.bookings.map(booking =>
+          ({
+            id: booking.booking_id,
+            title: booking.booking_name,
+            start: booking.start_time,
+            end: booking.end_time,
+            resourceId: booking.room_id,
+            color: booking.room.color
+
+          })
+        )
+      },
+      resources() {
+        return this.rooms.map(room =>
+          ({
+            id: room.room_id,
+            title: room.room_name,
+            eventColor: room.color
+          })
+        )
+      },
+      configuration() {
+        let configuration = this.config
+        configuration.resources = this.resources
+        return configuration
+      },
+    },
+    methods: {
+      ...mapActions(['getBookings', 'getRooms', 'getExams']),
+      ...mapMutations([
+        'toggleBookRoomModal',
+        'setCalendarView'
+      ]),
+      eventRender(event, element, view) {
+        if (view.name === 'listWeek' || view.name === 'listDay') {
+        let id = event.resourceId
+          let eventTitle = event.title
+          let resTitle = this.resources.find(res => res.id == id).title
+          let examObj = this.exams.find(exam => exam.booking_id == id)
+          let invigilator = 'none'
+          if (examObj && examObj.invigilator&& examObj.invigilator.invigilator_name) {
```

Comment:
Formatting

Reply:
not sure what formatting issues you have here.  Please elaborate.

Reply:
Line 114 to 115 the “let” is not aligned. Something is askew